forest/src/cli/chain_cmd.rs

```diff
@@ -0,0 +1,125 @@
+    #[structopt(about = "<CID> Read the raw bytes of an object")]
+    ReadObj {
+        #[structopt(short, help = "Input a valid CID")]
+        cid: String,
+    },
+}
+
+impl ChainCommands {
+    pub async fn run(&self) {
+        // TODO handle cli config
+        match self {
+            Self::Block { cid } => {
+                let cid: Cid = cid.parse().unwrap();
+                let mut client = new_client();
+
+                let blk = block(&mut client, cid)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!("{}", serde_json::to_string_pretty(&blk).unwrap());
+            }
+            Self::Genesis => {
+                let mut client = new_client();
+
+                let gen = genesis(&mut client)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!("{}", serde_json::to_string_pretty(&gen).unwrap());
+            }
+            Self::Head => {
+                let mut client = new_client();
+
+                let canonical = head(&mut client)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!(
+                    "{}",
+                    serde_json::to_string_pretty(&canonical.0.cids()).unwrap()
+                );
+            }
+            Self::Message { cid } => {
+                let cid: Cid = cid.parse().unwrap();
+                let mut client = new_client();
+
+                let msg = messages(&mut client, cid)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!("{}", serde_json::to_string_pretty(&msg).unwrap());
+            }
+            Self::ReadObj { cid } => {
+                let cid: Cid = cid.parse().unwrap();
+                let mut client = new_client();
+
+                let obj = read_obj(&mut client, cid)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
+                println!("{}", serde_json::to_string_pretty(&obj).unwrap());
+            }
+        }
+    }
+}
+
+fn stringify_rpc_err(e: JsonRpcError) -> String {
+    match e {
+        JsonRpcError::Full {
+            code,
+            message,
+            data: _,
+        } => {
+            return format!("JSON RPC Error: Code: {} Message: {}", code, message);
+        }
+        JsonRpcError::Provided { code, message } => {
+            return format!("JSON RPC Error: Code: {} Message: {}", code, message);
+        }
+    }
+}
```

Comment:
Since the other commands also return RPC errors, i think it makes sense to move this to forest/src/cli/mod.rs instead of keeping it in this file